### PR TITLE
🌱 Install CNI with ClusterResourceSet

### PR DESCRIPTION
**Pull request by mboukhalfa**
**What this PR does / why we need it**:
This PR installs CNI using ClusterResourceSet

### CODE REVIEW

**Comment by kashifest** [comment]
You are not exporting CNI path anywhere?

**Comment by mboukhalfa** [comment]
> You are not exporting CNI path anywhere?

It is exported in the config file with CNI var

File: test/e2e/data/infrastructure-metal3/bases/cluster/crs.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+# ConfigMap object referenced by the ClusterResourceSet object and with
+# the CNI resource defined in the test config file
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "cni-${CLUSTER_NAME}-crs-0"
+data: ${CNI_RESOURCES}
```
Comment (kashifest) [inline test/e2e/data/infrastructure-metal3/bases/cluster/crs.yaml:8]: where are you defining this var `CNI_RESOURCES `?

**Review by kashifest — COMMENTED** [review]
/approve
/hold
@mboukhalfa please unhold it when you have done all the logs check as discussed offline.

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

File: test/e2e/data/infrastructure-metal3/bases/cluster/crs.yaml (hunk 1 above)
Comment (lentzi90) [reply test/e2e/data/infrastructure-metal3/bases/cluster/crs.yaml:8]: For future reference, the `e2e_cni.CNIPath` is the same as `CNI` in the e2e config. See https://github.com/kubernetes-sigs/cluster-api/blob/fa70f61a77828cea2d5d059ffd0dfcf6bcbc9f2d/test/e2e/common.go#L39

**Pull request merged**